File: problems/quest/data_structures_and_algorithms/arrays/Build_an_Array_With_Stack_Operations.py
class BuildAnArrayWithStackOperations:
    def buildArray(self, target: list[int], n: int) -> list[str]:
        ans = []
        stack = [0] * len(target)

        idx = 0
        for num in range(1, n + 1):
            stack[idx] = num
            ans.append("Push")

            if num != target[idx]:
                ans.append("Pop")
            else:
                idx += 1
                if idx == len(target):
                    break

        return ans

File: problems/quest/data_structures_and_algorithms/arrays/BuildAnArrayWithStackOperations.java
import java.util.ArrayList;
import java.util.List;

public class BuildAnArrayWithStackOperations {
    public List<String> buildArray(int[] target, int n) {
        var ans = new ArrayList<String>();
        int[] stack = new int[target.length];

        for (int num = 1, idx = 0; num <= n && idx < target.length; num++) {
            stack[idx] = num;
            ans.add("Push");

            if (num != target[idx])
                ans.add("Pop");
            else
                idx++;
        }

        return ans;
    }
}
